recipe/meta.yaml

```diff
@@ -24,6 +24,7 @@ test:
     - docformatter 1.7.*
     - isort 5.12.*
     - jq 1.6.*
+    - make >=3.8
```

Comment:
I think this might have been the root cause of the issue we were seeing.

When `condev` is installed in the `base` conda environment, it installs `make` in that environment. Since it's that base environment from which `make package` runs (used in other CI scripts), `make` is available.

But when we run `make format`, we're in a _development environment_, where `make` is not installed. On other systems -- like our personal systems, for example, or HPC systems -- `make` is almost certainly available from the operating system. But it may well not be installed in a pristin container that CI jobs run in. I'm not certain that's the case with the containers that GitHub uses, but when I did my own local tests, it was indeed a problem.

By adding `make` to the `uwtools` test requirements, we ensure that it is available in development environments / shells. And this is a good idea, because we expect developers in dev shells to run e.g. `make format` and `make test`. Usually they've already got `make` on their systems, but we shouldn't assume so, and should supply whatever we need. That's what this new run requirement does.